src/stdlib_strings_to_string.fypp

```diff
@@ -0,0 +1,123 @@
+#:include "common.fypp"
+submodule(stdlib_strings) stdlib_strings_to_string
+
+    integer, parameter :: buffer_len = 128
+    character(len=*), parameter :: err_sym = "[*]"
+        !!TODO: [*]?
+
+contains
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    !> Format or transfer a ${t1}$ scalar as a string.
+    pure module function to_string_${t1[0]}$_${k1}$(value, format) result(string)
+        ${t1}$, intent(in) :: value
+        character(len=*), intent(in), optional :: format
+        character(len=:), allocatable :: string
+        
+        character(len=buffer_len) :: buffer
+        integer :: stat
+
+        write(buffer, optval(format, "(g0)"), iostat=stat) value
```

Comment:
@St-Maxwell @zoziha @awvwgk 

I apologize for the last minute suggestion. What do you think about the following:

```suggestion
        write(buffer, "(" // optval(format, "g0") // ")", iostat=stat) value
```

Whether the format dummy is present or not, this encloses the format string in an extra set of parentheses. This makes the function slightly more flexible. For example, you can now do:

```
print *, to_string(1.2345, 'f4.2') ! prints '1.23'
```

which would previously not work. I don't know why Fortran requires parentheses around format strings, but it's always been one of those little annoying things to me. And if you already have an outer set of parentheses, the semantics don't change. I think this allows for a better UI.

Reply:
I have to admit, a format string without parenthesis looks slightly off to me. But I have no strong feelings about this either way.

The only issue I could see, is that it might be surprising for a new user if the intrinsic transfer routines require a different syntax than the standard library. But maybe this is a good thing as well and could be used to suggest a simplification of format strings in the next Fortran standard.

Reply:
Yes, if we decide to do this we should document it clearly.

Reply:
Without parentheses, it looks more concise. I think it's good, as long as we explain it in the document: We use parentheses by default inside `to_string` to help users write less.
I will confirm the feasibility of this writing later.

Reply:
I just made a suggestion in https://github.com/fortran-lang/stdlib/pull/444#discussion_r693386425.